common/src/api/external/mod.rs

```diff
@@ -1433,11 +1433,22 @@ impl SimpleIdentityOrName for AntiAffinityGroupMember {
 
 // DISKS
 
-#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
-#[serde(rename_all = "snake_case")]
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(tag = "type", rename_all = "snake_case")]
 pub enum DiskType {
-    Distributed,
-    Local,
+    Distributed {
+        /// ID of snapshot from which disk was created, if any
+        snapshot_id: Option<Uuid>,
+        /// ID of image from which disk was created, if any
+        image_id: Option<Uuid>,
+    },
+
+    Local {
+        /// ID of the sled this local disk is allocated on, if it has been
+        /// allocated. Once allocated it cannot be changed or migrated.
+        #[schemars(with = "Option<Uuid>")]
+        sled_id: Option<SledUuid>,
```

Comment:
What's the utility of this? I'm always loth to give users data that they can *only* misuse.